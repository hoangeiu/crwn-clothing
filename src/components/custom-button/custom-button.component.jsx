import React from "react";

import "./custom-button.styles.scss";

const CustomeButton = ({
  children,
  isGoogleSignIn,
  inverted,
  ...otherProps
}) => (
  <button
    className={`${inverted ? "inverted" : ""} ${
      isGoogleSignIn ? "google-sign-in" : ""
    } custom-button`}
    {...otherProps}
  >
    {children}
  </button>
);

export default CustomeButton;
