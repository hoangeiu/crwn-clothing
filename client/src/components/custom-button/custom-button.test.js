import React from "react";
import { shallow } from "enzyme";

import { CustomButton } from "./custom-button.component";

describe("CustomButton component", () => {
  let wrapper;
  let mockChildren;
  let mockiIsGoogleSignIn;
  let mockInverted;
  beforeEach(() => {
    mockChildren = "Sign in with Google";
    mockiIsGoogleSignIn = false;
    mockInverted = false;

    const mockProps = {
      children: mockChildren,
      isGoogleSignIn: mockiIsGoogleSignIn,
      inverted: mockInverted,
    };

    wrapper = shallow(<CustomButton {...mockProps} />);
  });

  it("should CustomButton renders without error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should has inverted class when inverted prop is true", () => {
    mockInverted = true;

    wrapper = shallow(<CustomButton {...{ inverted: mockInverted }} />);

    expect(wrapper.exists(".inverted")).toBeTruthy;
  });

  it("should has google-sign-in class when isGoogleSignIn prop is true", () => {
    wrapper = shallow(<CustomButton {...{ isGoogleSignIn: true }} />);

    expect(wrapper.exists(".google-sign-in")).toBeTruthy();
  });

  it("should render children", () => {
    expect(wrapper.find("button").text()).toBe(mockChildren);
  });
});
