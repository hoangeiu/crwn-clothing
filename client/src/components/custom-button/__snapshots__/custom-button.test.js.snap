// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton component should CustomButton renders without error 1`] = `
<button
  className="  custom-button"
>
  Sign in with Google
</button>
`;
